contracts/TestContracts.sol

```diff
@@ -88,3 +89,22 @@ contract TestForwarder {
         require(success, "Relay call failed");
     }
 }
+
+contract TestTarget {
+    uint public calls = 0;
+    uint public gasLeft;
+    bytes public data;
+    fallback() external payable {
+        calls++;
+        data = msg.data;
+        gasLeft = gasleft();
```

Comment:
Better save gasleft() first, otherwise it is affected by the previous writes and its value becomes much smaller.